### ISSUE TITLE
nested_tabs - initialize `data` with `JoinKeys` object 

### ISSUE THREAD

**Issue by gogonzo** [insightsengineering/teal#683]
Initialize `data` object with `JoinKeys` object included.

Blocked by: 
- https://github.com/insightsengineering/teal.slice/issues/63

**Comment by nikolas-burkoff** [comment]
Make sure we handle merge with the join keys appropriately - either here or in another issue

**Comment by nikolas-burkoff** [comment]
Given where `$get_join_keys()` is used:

![Image](https://user-images.githubusercontent.com/15201933/179237457-d388201f-4d36-4a65-9523-e69b6133522e.png)

Proposal:

- Replace `FilteredData$get_join_keys(d1, d2)` with `FilteredData$get_join_keys()` which returns the whole JoinKeys object. This does mean that if you want to get explicit join_key pairs you would need to call `FilteredData$get_join_keys()$get(d1, d2)` but this is simple enough to use in CDISCFilteredData where it is used
- Inside nested tabs use `attr(data, "join_keys") <- datasets$get_join_keys()` (though see note below)
- Inside teal.transform update extract and merge to use the JoinKeys object not the list where needed

Notes:

- At the moment we use `attr(data, "join_keys") <- datasets$get_join_keys()[datanames]` - we can either not worry about this filtering step and pass the whole join_keys in OR we can have a method on JoinKeys for filtering i.e. `datasets$get_join_keys().filter(datanames)` which returns a new JoinKeys object which only contains keys for datasets whose names are in `datanames`

- We should make sure we are handling empty/NULL join_keys appropriately

**Comment by gogonzo** [comment]
I agree passing `JoinKeys` makes more sense. We can register `[[`, `[`, `$` methods for `JoinKeys` to return a list for a given datanames so `datasets$get_join_keys()$get()[datanames]` won't bother anyone in the future.

**Comment by nikolas-burkoff** [comment]
> I agree passing JoinKeys makes more sense. We can register [[, [, $ methods for JoinKeys to return a list for a given datanames so datasets$get_join_keys()$get()[datanames] won't bother anyone in the future.

I've added this to here https://github.com/insightsengineering/teal.data/issues/78
